docs-src/http-routing.md

```diff
@@ -0,0 +1,63 @@
+# HTTP routing
+
+HTTPRoute resources allow you to match on HTTP traffic and direct it Kubernetes
+backends. This guide shows how the HTTPRoute matches on host, header, and path
+fields and forwards traffic to different Kubernetes Services.
+
+The following diagram describes a required traffic flow across three different
+Services:
+
+- Traffic to `foo.example.com/login` is forwarded to `foo-svc` 
+- Traffic to `bar.example.com/*` with a `Env: canary` header is forwarded 
+to `bar-svc-canary`
+- Traffic to `bar.example.com/*` without the header is forwarded to `bar-svc` 
+- Any remaining non-matching traffic is responded to with an HTTP 404 message
+
+![HTTP Routinng](./images/http-routing.png)
+
+The dotted lines show the Gateway resources deployed to configure this routing
+behavior. There are two HTTPRoute resources that create routing rules on the
+same `prod-web` Gateway. This illustrates how [more than one Route can bind to a
+Gateway](api-overview.md/#gateway) which allows Routes to merge on a Gateway as
+long as they don't conflict. 
+
+The following `prod-web` Gateway is defined from the `acme-lb` GatewayClass.
+`prod-web` listens for HTTP traffic on port 80 and will bind to all Routes in
+the `default` Namespace that have the matching labels  `gateway: prod-web-gw`.
+Route labels allow the Route to determine which Gateways recieve their routing
```

Comment:
Although technically true, this feels a bit backwards to me. Gateways select Routes with specific labels, Route owners can explicitly allow from specific Gateways using the `gateways` field on a route. I think there's a few different ways users could choose to make this connection, and we probably need a doc just for this discussion. For example:

1. Routes are labeled with `app` and `env` labels, Gateway selects based on app and/or env it is intended to serve traffic for.
2. Gateways select Routes with a specific label like `prod-web-gw`, Routes add label to indicate the Gateway they'd like to be served by.
3. Gateways select all routes in the namespace, Routes only allow selection from a specific Gateway or set of Gateways.

I don't really like 3, but it is technically possible. I think I prefer 1 over 2 as it more closely matches the intent of the API, but I can see the use case for 2. So with all that said, I'm not quite sure how to phrase this sentence, I'm just hesitant to frame it so concretely since this is one of several possible models for connecting Gateways and Routes.

Reply:
Since matching Routes with Gateways (or Gateways with Routes) has so much flexibility I agree we need a section (probably in api-overview) that talks about conventions for connecting the two. 

Personally, I find it really strange not to be more explicit around which gateway is being used, but this is another thing we need to get feedback from users from on how they think about connecting Gateways and Routes. I'll tone down this sentence to make it a little less suggestive about which is the right convention.